onboard/Joint.hpp

```diff
@@ -0,0 +1,187 @@
+#ifndef JOINT_HPP_
+#define JOINT_HPP_
+#include <EEPROM.h> //ROM memory library
+#include "ROM_DATA.h" // array to data addresses and enum
+
+const int zeroTorque = 127; //PWM value for zero torque
+const int torqueOffset = 102; //max offset from zero torque
+const int minPWM = zeroTorque-torqueOffset; //27
+const int maxPWM = zeroTorque+torqueOffset; //229
+const int ScaleFactor = 1; // in case we need it
+const int OutOfRangeThreshold = 4;
+const int minPotNaturalRange = 10;
+const int maxPotNaturalRange = 1014;
+
+class Joint {
+  public:
+    //Constructors
+    Joint() {};
+    Joint(int jointNumber_, int potPin_, int motorPin_, int enablePin_, float kP_, float kI_, float kD_, int direction_) :
+      jointNumber(jointNumber_),
+      potPin(potPin_),
+      motorPin(motorPin_),
+      enablePin(enablePin_),
+      kP(kP_),
+      kI(kI_),
+      kD(kD_),
+      direction(direction_) {}
+
+    //Setters
+    void setMinPot(int minPot_){minPot = minPot_;}
+    void setMaxPot(int maxPot_){maxPot = maxPot_;}
+    void setMinTheta(int minTheta_){minTheta = minTheta_;} // in rads values
+    void setMaxTheta(int maxTheta_){maxTheta = maxTheta_;} // in rads values
+    void setZeroPot(int zeroPot_){zeroPot = zeroPot_;}
+    void setLocalJointNum(){localJointNum = ((jointNumber-1)%numOfJoints);}//get number from 0 to 2 instead of 1-12
+    void setMemoryAddr(JointROM memoryAddr_){memoryAddr = memoryAddr_;}
+
+    //Getters
+    int getJointNumber() {return jointNumber;}
+    int getPotPin() {return potPin;}
+    int getMotorPin() {return motorPin;}
+    int getEnablePin() {return enablePin;}
+    float getkP () {return kP;}
+    float getkI () {return kI;}
+    float getkD () {return kD;}
+    int getDirection () {return direction;}
+    int getMotorOrientation() {return motorOrientation;}
+    int getPotOrientation() {return potOrientation;}
+    int getPotCabling() {return potCabling;}
+    int getMinPot() {return minPot;}
+    int getMaxPot() {return maxPot;}
+    int getMinTheta() {return minTheta;}
+    int getMaxTheta() {return maxTheta;}
+    int getZeroPot() {return zeroPot;}
+    int getLocalJointNum() {return localJointNum;}
+    JointROM getMemoryAddr(){return memoryAddr;}
+
+    //Other methods
+    int readPotentiometer(){return analogRead(potPin);}
+    void motorPWM(int drive){analogWrite(motorPin,drive);}
+    void setEnable(bool state){digitalWrite(enablePin, state);}
+    void setSetPointFromPot(){setPoint = readPotentiometer();}
+    int readROM_joint(){
+      uint16_t val;
+      EEPROM.get(memoryAddr.jointAddr,val);
+      return val;
+    }
```

Comment:
Nontrivial method definitions (ie not getters and setters) should be in a seperate .cpp file instead of in the header file